app/controllers/characters_controller.rb

```diff
@@ -0,0 +1,58 @@
+class CharactersController < ApplicationController
+  before_action :current_resource_owner
+  before_action :current_resource, only: [:show, :edit, :update]
+  before_action :only_validated_character, only: :show
+  before_action :only_denied_character, only: [:edit, :update]
+
+  def index
+    @characters = @user.characters.includes(:user)
+  end
+
+  def new
+    @character = @user.characters.new
+  end
+
+  def create
+    @character = @user.characters.new(character_params)
+    if @character.save
+      redirect_to user_characters_path(@user)
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @character.update(character_params)
+      if @character.validating?
```

Comment:
When character is in validating state user not available to move in character show page, so he will be redirected to index. It happens when user Edit application after denied.